src/adapter/src/catalog.rs

```diff
@@ -1395,20 +1395,52 @@ impl ExprHumanizer for ConnCatalog<'_> {
     }
 
     fn column_names_for_id(&self, id: GlobalId) -> Option<Vec<String>> {
-        self.state
-            .entry_by_id
-            .get(&id)
-            .try_map(|entry| {
-                Ok::<_, SqlCatalogError>(
-                    entry
-                        .desc(&self.resolve_full_name(entry.name()))?
-                        .iter_names()
-                        .cloned()
-                        .map(|col_name| col_name.to_string())
-                        .collect(),
-                )
-            })
-            .unwrap_or(None)
+        let Some(entry) = self.state.entry_by_id.get(&id) else {
+            return None;
+        };
+
+        match entry.index() {
+            Some(index) => {
+                let Some(on_entry) = self.state.entry_by_id.get(&index.on) else {
+                    return None;
+                };
+                let Ok(on_desc) = on_entry.desc(&self.resolve_full_name(on_entry.name())) else {
+                    return None;
+                };
+
+                let mut on_names = on_desc
+                    .iter_names()
+                    .map(|col_name| col_name.to_string())
+                    .collect::<Vec<_>>();
+
+                let (p, _) = mz_expr::permutation_for_arrangement(&index.keys, on_desc.arity());
+
+                // Init ix_names with unknown column names.
+                let ix_arity = p.keys().cloned().max().expect("max key") + 1;
```

Comment:
> Replaced this with `.unwrap_or(0)` which avoids the panic in the above case.

Shouldn't it be `.unwrap_or(-1)`, strictly speaking? Because there is a +1 afterwards.

Reply:
Can't do that because `-1` is not `usize`. I did `.map(|m| m + 1).unwrap_or(0);` instead for consistency with the relation arity.